app/src/ingestion/pdf_postprocess.py

```diff
@@ -1,18 +1,67 @@
 import logging
 
 from src.ingestion.pdf_elements import EnrichedText, TextType
+from src.ingestion.pdf_stylings import Styling
+from src.util.string_utils import basic_ascii
 
 logger = logging.getLogger(__name__)
 
 
+def associate_stylings(
+    enriched_texts: list[EnrichedText], stylings: list[Styling]
+) -> list[EnrichedText]:
+    "Given EnrichedTexts and Stylings, assocate stylings to the corresponding text item"
+    for e_text in enriched_texts:
+        matched_stylings = [
+            styling for styling in stylings if styling_matches_text(styling, e_text)
+        ]
+        if matched_stylings:
+            e_text.stylings = matched_stylings
+    return enriched_texts
+
+
+def styling_matches_text(styling: Styling, e_text: EnrichedText) -> bool:
+    # Quick checks
+    if styling.pageno != e_text.page_number or styling.headings != e_text.headings:
+        return False
+
+    # Slower checks
+    stripped_wider_text = basic_ascii(styling.wider_text).strip()
+    stripped_e_text = basic_ascii(e_text.text).strip()
```

Comment:
What's the purpose of stripping to ASCII? Rendering differences between pdfminer and unstructured?

Reply:
pdfminer extracts the bullet (`• `) from the PDF, e.g. https://github.com/navapbc/labs-decision-support-tool/blob/0323b304eee9232483f76bd8bb2067fcede9b082/app/tests/src/ingestion/test_pdf_stylings.py#L17

unstructured removes the bullet.